### PR TITLE
Rust 过程宏初学者手册

### LINKED ISSUE #419: [zh]（Rust 过程宏初学者手册）
[Procedural Macros in Rust – A Handbook for Beginners](https://www.freecodecamp.org/news/procedural-macros-in-rust/)

### PR DESCRIPTION
**Pull request by hezean**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #419.

### CODE REVIEW

File: articles/zh/procedural-macros-in-rust.md [hunk 1]
```diff
@@ -1167,19 +1175,19 @@ version = "0.1.0"
 edition = "2021"
 resolver = "2"
 
-# 更多键及其定义请参见 https://doc.rust-lang.org/cargo/reference/manifest.html
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (miyaliu666) [inline articles/zh/procedural-macros-in-rust.md:1178]:
```suggestion
# 更多键及其定义请参见  https://doc.rust-lang.org/cargo/reference/manifest.html
```

File: articles/zh/procedural-macros-in-rust.md [hunk 2]
```diff
@@ -1799,8 +1802,8 @@ impl Parse for ParsedMap {
     fn parse(input: ParseStream) -> syn::Result<Self> {
         let mut entries = Vec::<ParsedMapEntry>::new();
 
-        // 检查输入是否为空（没有参数传递）。如果不是，则
-        // 报错，因为我们无法继续下去。
+        // 检查输入是否为空（没有参数传递）。
+        // 如果不是，则报错，因为我们无法继续下去。
         if input.is_empty() {
             panic!("至少要为一个空的 hashmap 指定一个类型");
```
Comment (miyaliu666) [inline articles/zh/procedural-macros-in-rust.md:1808]:
```suggestion
            panic!("At least a type must be specified for an empty hashmap");
```

File: articles/zh/procedural-macros-in-rust.md [hunk 3]
```diff
@@ -1828,23 +1829,19 @@ impl Parse for ParsedMap {
                 panic!("键必须是字符串字面量或标识符！");
```
Comment (miyaliu666) [inline articles/zh/procedural-macros-in-rust.md:1829]:
```suggestion
                panic!("Key must be either a string literal or an identifier!");
```

File: articles/zh/procedural-macros-in-rust.md [hunk 4]
```diff
@@ -1828,23 +1829,19 @@ impl Parse for ParsedMap {
...
-                // 对于数字，`false` 对于布尔值等。
+                // 如果输入不是标识符，尝试解析它作为字面值，
+                // 比如 `"string"` 是字符串，`42` 是数字，`false` 是布尔值等。
             } else if let Ok(value) = input.parse::<Lit>() {
                 value.to_token_stream()
             } else {
-                // 如果输入既不是标识符，也不是字面值，
-                // 则报错。
+                // 如果输入既不是标识符，也不是字面值，则报错。
                 panic!("值必须是字面量或标识符！");
```
Comment (miyaliu666) [inline articles/zh/procedural-macros-in-rust.md:1845]:
```suggestion
                panic!("Value must be either a literal or an identifier!");
```

**Review by miyaliu666 — CHANGES_REQUESTED** [review]

File: articles/zh/procedural-macros-in-rust.md (hunk 1 above)
Comment (hezean) [reply articles/zh/procedural-macros-in-rust.md:1178]: this line is generated by `cargo new` command, not by the author

**Comment by miyaliu666** [comment]
谢谢 @hezean 非常详细地校对这篇长译文，还添加了“译者注”。我提出一些修改建议，供你参考：）

File: articles/zh/procedural-macros-in-rust.md (hunk 2 above)
Comment (hezean) [reply articles/zh/procedural-macros-in-rust.md:1808]: these are just prompts that will be printed, for chinese readers' better understanding i guess it's okay to translate them? wdyt

File: articles/zh/procedural-macros-in-rust.md (hunk 1 above)
Comment (miyaliu666) [reply articles/zh/procedural-macros-in-rust.md:1178]: Thank you for explaining. So it shouldn't be translated.

File: articles/zh/procedural-macros-in-rust.md (hunk 2 above)
Comment (miyaliu666) [reply articles/zh/procedural-macros-in-rust.md:1808]: Great. Let's translate them.

**Pull request merged**